test/linalg/matmul.jl

```diff
@@ -5,189 +5,279 @@ using Base.Test
 ## Test Julia fallbacks to BLAS routines
 
 # matrices with zero dimensions
-for i = 1:10
```

Comment:
I don't remember why I did that (a comment would probably have helped), but my guess is what @tkelman said. When multiplying an mx0 matrix with an 0xn then we have a fast branch that just allocates an mxn matrix and sets the elements to zero. Repeating the tests makes it more likely to catch it if we forgot to set the memory to zero.